core/embed/trezorhal/optiga/optiga_transport.c

```diff
@@ -137,11 +137,32 @@ static size_t sec_chan_size = 0;
 #ifdef NDEBUG
 #define OPTIGA_LOG(prefix, data, data_size)
 #else
+#include <stdio.h>
 static optiga_log_hex_t log_hex = NULL;
 void optiga_set_log_hex(optiga_log_hex_t f) { log_hex = f; }
-#define OPTIGA_LOG(prefix, data, data_size) \
-  if (log_hex != NULL) {                    \
-    log_hex(prefix, data, data_size);       \
+#define OPTIGA_LOG(prefix, data, data_size)                                  \
+  if (log_hex != NULL) {                                                     \
+    static uint8_t prev_data[4];                                             \
+    static size_t prev_size = 0;                                             \
+    static int repeated = 0;                                                 \
+    if (prev_size == data_size && memcmp(data, prev_data, data_size) == 0) { \
+      repeated += 1;                                                         \
+    } else {                                                                 \
+      if (repeated > 0) {                                                    \
+        char ext_prefix[25];                                                 \
+        snprintf(ext_prefix, sizeof(ext_prefix), "%sREPEATED (%d) ", prefix, \
```

Comment:
fwiw we have a `mini_snprintf` that would be a better choice here if we want this code to go master repo

Reply:
I suppose we don't need to show the exact count. Also it's hard to write it out clearly. So I simplified it https://github.com/trezor/trezor-firmware/pull/3296/commits/286acf0e7b9eb936074d154e4f83d98af4f59abf.